### PR TITLE
chore(EA-1942): Run acceptance tests for a WID token in Input API build

### PR DESCRIPTION
**Pull request by TurcanStanislav**

### CODE REVIEW

File: .github/workflows/pnp-input-api-build.yml [hunk 1]
```diff
@@ -316,12 +316,21 @@ jobs:
               "apikey": "${{ env.NUGET_API_KEY }}"
             }]
 
+      - name: Get Identity token
+        id: identitytoken
+        uses: extenda/actions/identity-token@v0
+        with:
+          service-account-key: ${{ secrets.GCLOUD_AUTH_STAGING }}
+          service-account: braveheart-quotes-webclient@quotes-staging-ccdf.iam.gserviceaccount.com
+          audiences: bhq-braveheart-quotes
```
Comment (nadiiaboichuk) [inline .github/workflows/pnp-input-api-build.yml:325]: The audience should be passed as input parameter as it should be different for all Input APIs

File: .github/workflows/pnp-input-api-build.yml [hunk 2]
```diff
@@ -316,12 +316,21 @@ jobs:
               "apikey": "${{ env.NUGET_API_KEY }}"
             }]
 
+      - name: Get Identity token
+        id: identitytoken
+        uses: extenda/actions/identity-token@v0
+        with:
+          service-account-key: ${{ secrets.GCLOUD_AUTH_STAGING }}
+          service-account: braveheart-quotes-webclient@quotes-staging-ccdf.iam.gserviceaccount.com
```
Comment (nadiiaboichuk) [inline .github/workflows/pnp-input-api-build.yml:324]: Not sure we can use this test service account email here

File: .github/workflows/pnp-input-api-build.yml [hunk 3]
```diff
@@ -316,12 +316,21 @@ jobs:
...
+          audiences: bhq-braveheart-quotes
+
       - name: Run Acceptance Tests
         run: |
           dotnet test ${{ secrets.path-to-acceptance-tests }}
         env:
           INPUT_API_URL: ${{ secrets.staging-service-base-url }}
           BEARER_TOKEN: ${{ steps.testtoken.outputs.iam-token }}
+          IDENTITY_TOKEN: ${{ steps.identitytoken.outputs.identity-token }}
```
Comment (nadiiaboichuk) [inline .github/workflows/pnp-input-api-build.yml:333]: Maybe it makes sense to run the same set of tests with WID token? Or how did you plan to use this env variable in the acceptance tests?

File: .github/workflows/pnp-input-api-build.yml (hunk 2 above)
Comment (TurcanStanislav) [reply .github/workflows/pnp-input-api-build.yml:324]: Ok, which we can use instead? Or whom we should ask about it? Platform team?

File: .github/workflows/pnp-input-api-build.yml (hunk 3 above)
Comment (TurcanStanislav) [reply .github/workflows/pnp-input-api-build.yml:333]: Ok

File: .github/workflows/pnp-input-api-build.yml [hunk 4]
```diff
@@ -67,7 +67,17 @@ on:
         type: boolean
         required: false
         default: false
-        
+      service-name:
```
Comment (nadiiaboichuk) [inline .github/workflows/pnp-input-api-build.yml:70]: We have to move this parameter to secrets. 
It should not be required.
Let's use specific names like `wid-token-audience` and change the description to `The audience of the generated Workload Identity token. Should be the name of the receiver service.`.

File: .github/workflows/pnp-input-api-build.yml [hunk 5]
```diff
@@ -67,7 +67,17 @@ on:
...
+          description: |
+            The name of the service that will be used for generating WID token.
+          type: string
+          required: true
+      service-account:
```
Comment (nadiiaboichuk) [inline .github/workflows/pnp-input-api-build.yml:75]: We have to move this parameter to secrets. 
It should not be required.
Let's use specific names like `wid-token-service-account` and change the description to `The service account of the requestor service. Will be used during Workload Identity token generation.`.

File: .github/workflows/pnp-input-api-build.yml [hunk 6]
```diff
@@ -289,6 +299,14 @@ jobs:
           service-account-key: ${{ secrets.GCLOUD_AUTH_STAGING }}
           api-key: ${{ env.API_KEY }}
 
+      - name: Get Identity token
```
Comment (nadiiaboichuk) [inline .github/workflows/pnp-input-api-build.yml:302]: This step should be run only if two relevant secrets are provided

File: .github/workflows/pnp-input-api-build.yml [hunk 7]
```diff
@@ -316,12 +334,19 @@ jobs:
...
 
-      - name: Run Acceptance Tests
+      - name: Run Acceptance Tests with IAM token
         run: |
           dotnet test ${{ secrets.path-to-acceptance-tests }}
         env:
           INPUT_API_URL: ${{ secrets.staging-service-base-url }}
           BEARER_TOKEN: ${{ steps.testtoken.outputs.iam-token }}
+      
+      - name: Run Acceptance Tests with WID token
```
Comment (nadiiaboichuk) [inline .github/workflows/pnp-input-api-build.yml:344]: This step should be run only if two relevant `wid-` secrets are provided

**Review by nadiiaboichuk — CHANGES_REQUESTED** [review]

**Review by nadiiaboichuk — APPROVED** [review]

**Pull request merged**